plugins/generic/metadataExport/MetadataExportPlugin.inc.php

```diff
@@ -0,0 +1,421 @@
+			define('METADATA_EXPORT_PLUGIN_NAME', $this->getName());
+			define('HANDLER_CLASS', 'MetadataExportHandler'); 
+			$this->import('MetadataExportHandler'); 
+			return true;
+		}
+		return false;
+	}
+	
+	/**
+	 * @see PKPPlugin::getTemplatePath()
+	 */
+	function getTemplatePath() {
+		return parent::getTemplatePath() . 'templates/';
+	}
+	
+	/**
+	 * @see PKPPlugin::getName()
+	 */
+	function getName() {
+		return 'MetadataExportPlugin';
+	}
+	
+	/**
+	 * @see PKPPlugin::getDisplayName()
+	 */
+	function getDisplayName() {
+		return __('plugins.generic.metadataExport.displayName');
+	}
+	
+	/**
+	 * @see PKPPlugin::getDescription()
+	 */
+	function getDescription() {
+		return __('plugins.generic.metadataExport.description');
+	}
+	
+	/**
+	 * @function getMetadataExportFormatName - Get the name of the metadata format.
+	 * Should be overridden by subclasses.
+	 * @return String
+	 */
+	function getMetadataExportFormatName() {
+		assert(false);
+	}
+	
+	/**
+	 * @function getFileExtension - Get the file extension of the export file.
+	 * Should be overridden by subclasses.
+	 * @return String
+	 */
+	function getFileExtension() {
+		assert(false);
+	}
+	
+	/**
+	 * @function getRootElement - Get the type of the XML's root node
+	 * Should be overridden by subclasses.
+	 * @return String
+	 */
+	function getRootElement() {
+		assert(false);
+	}
+	
+	/**
+	 * @function getFileContent
+	 * Should be overridden by subclasses.
+	 * @param $journal Journal Object
+	 * @param $articles Array of Article Objects
+	 * @return Mixed (String or XMLCustomWriter Object)
+	 */
+	function getFileContent($journal, $articles) {
+		assert(false);
+	}
+	
+	/**
+	 * @function getType - Identification as a reviewed article
+	 * @return String
+	 */
+	function getType() {
+		return __('rt.metadata.pkp.peerReviewed');
+	}
+	
+	/**
+	 * @function getVolumeAbbreviation - Abbreviation for the term "issue"
+	 * @return String
+	 */
+	function getVolumeAbbreviation() {
+		return __('issue.vol');
+	}
+	
+	/**
+	 * @function getTypeIdentifier - Identifier for publication type
+	 * @return String
+	 */
+	function getTypeIdentifier() {
+		return 'info:eu-repo/semantics/article';
+	}
+	
+	/**
+	 * @function getVersionIdentifier - Identifier for publication version
+	 * @return String
+	 */
+	function getVersionIdentifier() {
+		return 'info:eu-repo/semantics/publishedVersion';
+	}
+	
+	/**
+	 * @function getCopyrightStatement - get the Copyright statement
+	 * @param $copyrightHolder String
+	 * @param $copyrightYear Int
+	 * @return String
+	 */
+	function getCopyrightStatement($copyrightHolder, $copyrightYear) {
+		return __('submission.copyrightStatement', array('copyrightHolder' => $copyrightHolder, 'copyrightYear' => $copyrightYear));
+	}
+
+	/**
+	 * @function getFileName - Fill white spaces in the article/issue/journal name
+	 * @param $name String
+	 * @return String
+	 */
+	function getFileName($name) {
+		return preg_replace('/\s/', '-', $name);
```

Comment:
I don't think this will safely sanitize filenames. What about reserved names like `.` or `..` or something containing a slash? It's better, if possible, to avoid using a temporary file entirely and just generate and serve the contents up from memory. I might be missing something that motivates the use of a temporary file, though.